corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js

```diff
@@ -153,8 +152,14 @@ hqDefine("cloudcare/js/formplayer/menus/controller", function () {
 
         if (menuResponse.breadcrumbs) {
             menusUtils.showBreadcrumbs(menuResponse.breadcrumbs);
-            if (!appPreview && ((menuResponse.langs && menuResponse.langs.length > 1) || enablePrintOption)) {
-                menusUtils.showFormMenu(menuResponse.langs, initialPageData('lang_code_name_mapping'));
+            if (!appPreview) {
+                let isFormEntry = !menuResponse.queryKey;
+                if (isFormEntry) {
+                    menusUtils.showMenuDropdown(menuResponse.langs, initialPageData('lang_code_name_mapping'));
```

Comment:
For what it's worth, if you want to combine these two initializations, you could move the call to `initialPageData` into `showMenuDropdown` - it doesn't need to be here.  `menuResponse.langs` of course does, and I could also see a case made for keeping them separate so it's clear langs are only applicable to forms

Reply:
I didn't consider that, but I would prefer to explicitly keep it separate